sklearn/decomposition/_pca.py

```diff
@@ -210,6 +210,12 @@ class PCA(_BasePCA):
 
         .. versionadded:: 0.18.0
 
+    n_oversamples_rate : float, default=0.8
```

Comment:
This entry should be moved above `random_state`.

Reply:
I think that we should keep `n_oversamples` and set it to `10` to have the previous behaviour.
I don't know if we should support both a float and an integer. At least we need to bring all the information of the docstring of the `randomized_svd` to explain what is the impact of the parameter.

Reply:
I am agree with you that the explanation of the parameter is necessary. For the explanation of the type, I have also made a further explanation below, and I hope that there will be more discussions in this regard. 
Thanks.